nexus/types/src/external_api/views.rs

```diff
@@ -1488,6 +1488,22 @@ pub struct TargetRelease {
 
     /// The source of the target release.
     pub release_source: TargetReleaseSource,
+
+    /// Whether the system has been updated to a new release through the
+    /// recovery (MUPdate) path since the target release was last set.
+    ///
+    /// In this case, the target release is considered to be no longer active,
+    /// and a new release must be set.
+    pub mupdate_override: Option<TargetReleaseMupdateOverride>,
+}
+
+/// View of MUPdate override information for a target release.
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize, JsonSchema)]
+pub struct TargetReleaseMupdateOverride {
+    /// The minimum generation number required for the system to be back in
+    /// charge again.
+    pub minimum_generation: i64,
```

Comment:
I don't see what sense an external API consumer can make of this number.  We'd have to explain how we store the target release and what the generation number means, and we'd be way into implementation details.  I'd leave it out altogether.

Reply:
Ah I had it here beause I wanted to mirror the existing generation number that's already part of `views::TargetRelease`. There was also the other concern that I didn't just want to return `"mupdate_override": true` in the JSON, and I also didn't want to just return `"mupdate_override": {}`. In the future we could also add the time at which the minimum generation was set (though the schema doesn't have that yet).

Reply:
I didn't realize there was one there already.  I'm not sure that makes sense, though that one could at least tell you if the target release object you're looking at has changed since some previous request was made.

My suggestion would be to make this a boolean for now.  Each scheduled release is a major API bump so in principle we can change this easily.  I expect that eventually we may want an API that lists "mupdate overrides in place" and gives more details (like: this sled was mupdated, apparently at this time, apparently to this release (or some unknown release)).  Short of that, I don't think there's more than a boolean's worth of information to convey here.

Maybe @ahl has some thoughts here.

Reply:
I just have a bias towards untyped booleans I guess 😅

Reply:
Turned this into a bool for now.